### DISCUSSION TITLE
load_dict bsdf

### DISCUSSION THREAD

**Discussion by zichenwang01** [mitsuba-renderer/mitsuba3#1151] [Q&A] [answered]
Hi,
I am having a bit of trouble loading a custom bsdf. Here is my attempt
```
bsdf : mi.BSDF = mi.load_dict({
            'type': 'diffuse', 
            'reflectance':  mi.Color3f(0.8, 0.2, 0.2)
        })
```
and the problem I get
```
RuntimeError: ​[xml_v.cpp:341] Unkown value type: <class 'mitsuba.cuda_ad_rgb.Color3f'>
```

I looked at https://mitsuba.readthedocs.io/en/latest/src/others/bsdf_deep_dive.html but did not instructions for loading reflectance that has more than 1 number.

Thanks!

**Comment by njroussel** [comment, marked as answer]
Hi @zichenwang01 

The plugin pages ([for BSDF](https://mitsuba.readthedocs.io/en/latest/src/generated/plugins_bsdfs.html#smooth-diffuse-material-diffuse)) always have snippets of examples to initialize plugins from XML or Python dictionaries.

Here's what you want to be doing:

```python
bsdf : mi.BSDF = mi.load_dict({
            'type': 'diffuse',
            'reflectance':  {
                'type': 'rgb',
                'value': [0.8, 0.2, 0.2]
                #'value': mi.ScalarColor3f(0.8, 0.2, 0.2) <- This is also valid
            }
        })
```
